src/hooks/useNotificationSetup.ts

```diff
@@ -14,13 +29,42 @@ export const useNotificationSetup = () => {
     initNotifications()
   }, [])
 
+  const handleDeeplink = useCallback(
+    (deeplinkPath?: string) => {
+      if (!deeplinkPath) return
+
+      // Deeplink that is used in `linkTo` method after authorized checks
+      let finalDeeplink = deeplinkPath
+
+      const isAuthorizedPath = isAuthorizedLink(deeplinkPath)
+      if (!isSignedIn && isAuthorizedPath) {
+        AsyncStorage.setItem(ASYNC_STORAGE_KEYS.NEXT_DEEP_LINK, deeplinkPath)
+        finalDeeplink = '/sign-in'
+      }
+
+      const addSlash = !finalDeeplink.startsWith('/')
+      finalDeeplink = `${addSlash ? '/' : ''}${finalDeeplink}`
+
+      setDeeplink(finalDeeplink)
+      if (opts?.enableDeeplink) {
+        linkTo(finalDeeplink)
+      }
+    },
+    [linkTo, isSignedIn, opts?.enableDeeplink]
+  )
+
   // CONFIG: Handle in app notification
   useEffect(() => {
     if (notification) {
       Alert.alert('Notification', JSON.stringify(notification), [
         { text: 'Cancel', onPress: () => setNotification(undefined), style: 'cancel' },
         { text: 'Ok', onPress: () => setNotification(undefined) },
       ])
+
+      const deeplinkPath = notification.request.content.data.deeplink as string | undefined
```

Comment:
Prevent crashes
```suggestion
      const deeplinkPath = notification?.request?.content?.data?.deeplink as string | undefined
```

Reply:
I know types says something else but I remember that there is some edgecase when data can be undefined